### PR TITLE
WooCommerce Plugin:  Integration with Doppler App.

### PR DESCRIPTION
**Pull request by hernangalvan**
- Added class to handle integration with Doppler App, connection, disconnection and creation and submission of WC API Keys.
- Added class to save cart session to database when user adds or removes items from cart. 
- Added class to manage persistent admin notices (saving temporary message to a database field, deletes it after displaying).
- On first plugin synchronization creates the WC API keys and sends them to APP.
- When deactivating plugin, deletes integration. 
- When reactivating plugin creates new WC Keys and sends them to APP.

### CODE REVIEW

File: src/admin/js/doppler-for-woocommerce-admin.js [hunk 1]
```diff
@@ -77,16 +93,26 @@
 
 		syncListsButton.click(function(e){
 			e.preventDefault();
+			verifyKeys().then(syncrhonizeLists);
+		});
+
+		function verifyKeys(){
+			console.log('Verifying keys');
```
Comment (maurocardinali) [inline src/admin/js/doppler-for-woocommerce-admin.js]: should delete this console.log

File: src/admin/js/doppler-for-woocommerce-admin.js [hunk 2]
```diff
@@ -77,16 +93,26 @@
...
+			syncListsButton.attr('disabled','disabled').addClass("button--loading");
+			$("#dplr-settings-text").html(ObjWCStr.Synchronizing);
+			return $.post(ajaxurl, {action: 'dplrwoo_ajax_verify_keys'});
+		}
+
+		function syncrhonizeLists(resp){
+			console.log(resp.success); //wp_send_json_success or wp_send_json_error
```
Comment (maurocardinali) [inline src/admin/js/doppler-for-woocommerce-admin.js]: should delete this console.log

File: src/includes/class-doppler-for-woocommerce-app-connect.php [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+ * with the api to connect or disconnect
+ * the app integration.
+ *  
+ * @since 1.0.2
+ * @author Hernán Galván <hgalvan@makingsense.com>
+ */
+class Doppler_For_WooCommerce_App_Connect {
+
+	const INTEGRATION = 'magento';
+	const DEBUG_MODE = true;
```
Comment (maurocardinali) [inline src/includes/class-doppler-for-woocommerce-app-connect.php:15]: This should remain `true`?

**Review by maurocardinali — APPROVED** [review]
Awesome job Hernán! Just a few comments.

File: src/admin/js/doppler-for-woocommerce-admin.js (hunk 1 above)
Comment (hernangalvan) [reply src/admin/js/doppler-for-woocommerce-admin.js]: But i love my console logs :(

File: src/includes/class-doppler-for-woocommerce-app-connect.php (hunk 3 above)
Comment (hernangalvan) [reply src/includes/class-doppler-for-woocommerce-app-connect.php:15]: Yes! (for now), before production will delete this constant, for now its just a constant for using a testing api key and account wich is hardcoded... gonna remove later.

File: src/admin/js/doppler-for-woocommerce-admin.js (hunk 2 above)
Comment (hernangalvan) [reply src/admin/js/doppler-for-woocommerce-admin.js]: Done!

File: src/admin/js/doppler-for-woocommerce-admin.js (hunk 1 above)
Comment (hernangalvan) [reply src/admin/js/doppler-for-woocommerce-admin.js]: Done!

**Review by ghost — APPROVED** [review]
Excellent work Hernan!

**Pull request merged**